scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3961,14 +3961,21 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           syntaxError("expected '{' or indentation", at = token.pos)
         }
       } else {
-        (autoPos(Self(autoPos(Name.Anonymous()), None)), Nil)
+        syntaxError("expected 'with' <body>", at = token.pos)
```

Comment:
Removed `abstract givens cannot be annonymous` here since it's actually misleading. Givens with `with` cannot be abstract at all.